### DISCUSSION TITLE
Unable to use insights "onEvent" without getting duplicated view events

### DISCUSSION THREAD

**Discussion by Ali-hd** [algolia/instantsearch#6495] [Q&A] [answered]
Hey guys,

I am trying to forward my events to a third party and so I need to use the insights onEvent prop. There is currently an issue when I try to use it because I keep getting duplicated view events "Hits Viewed".

You can check the console and view the duplicated event when: 
- clicking on a facet like hierarchicalCategories or brand
- visiting a different page then navigating back to instant search page

Am I doing something wrong or is this a bug?

Here's a reproducible (I used app-router example)
https://codesandbox.io/p/devbox/cold-tdd-pmm5mq?workspaceId=ws_5V79nFTFYEgATD9vgxfVQP

**Comment by Haroenv** [comment, marked as answer]
I can reproduce this indeed. In the builtin event implementation that's not an issue as the events are debounced (so the same event doesn't get triggered multiple times). You could also implement that in your custom implementation

**Reply by Ali-hd** [reply]
I see, got it thanks
